client/core/core.go

```diff
@@ -491,6 +494,87 @@ func (dc *dexConnection) ack(msgID uint64, matchID order.MatchID, signable msgjs
 	return nil
 }
 
+// register submits a new 'register' request to the server. Depending on the
+// server version, either encKey or encKeyLegacy will be set, but not both.
+func (c *Core) register(dc *dexConnection, crypter encrypt.Crypter) (regRes *msgjson.RegisterResult,
+	acctPubB, encKey, encKeyLegacy []byte, paid bool, err error) {
+
+	creds := c.creds()
+
+	// register is called before the *dexConnection is in the Core.conns map,
+	// so no risk of concurrent access.
+	regV2 := dc.acct.dexPubKey != nil
+
+	var accountSuspended bool
+newkey:
+	for {
+		if regV2 && !accountSuspended {
+			encKey, acctPubB, err = dc.acct.setupCryptoV2(creds, crypter)
+			if err != nil {
+				return nil, nil, nil, nil, false, newError(acctKeyErr, "setupCryptoV2 error: %v", err)
+			}
+		} else {
+			encKey = nil
+			// This is an old-style DEX that doesn't provide their pubkey in the
+			// config response. Gotta use a legacy-style key.
+			encKeyLegacy, acctPubB, err = dc.acct.setupCryptoLegacy(crypter)
+			if err != nil {
+				return nil, nil, nil, nil, false, codedError(acctKeyErr, err)
+			}
+		}
+		dexReg := &msgjson.Register{
+			PubKey: acctPubB,
+			Time:   encode.UnixMilliU(time.Now()),
+		}
+		regRes = new(msgjson.RegisterResult)
+		err = dc.signAndRequest(dexReg, msgjson.RegisterRoute, regRes, DefaultResponseTimeout)
```

Comment:
The PR pretty well establishes using the `'register'` route to determine `AccountExistsError` or not, but this could be achieved with `'connect'` as well I believe, and that is something I am going to have to change in my `postbond` branch, which entirely deprecates the `'register'` route.  While I may not use `'connect'`, I will have to figure out account existence some other way.